### ISSUE TITLE
Problems when trying to count characters while using picker

### ISSUE THREAD

**Issue by BartoszGancza** [xLs51/Twemoji-Picker#11]
I am currently making a small web app as a final project for a programming course, and have encountered 2 potential problems when trying to count characters in the textarea, while using the picker.

1. There seems to be some conflicts while using jQuery key events to update the function. When I try to run it by `$("#twemoji-picker").keyup(function (){'function here'})` it does not pick up the event at all (there is absolutely no response). It starts working only after attaching the listener to `$(".twemoji-textarea")`, but then fails to update the counter after each key press (counter appears, but always stays at 140/140 count) if `$(this).val().length` is being used to get the string length. It then updates properly, if `$("#twemoji-picker").val().length` is used instead. This would be fine if someone only want the counter, but I also tried using substring to automatically cut the content of textarea after user exceeds the character limit, but to no avail (no combination of selectors for the substring seems to work).

2. I am a beginner so I don't really know what might be the reason of this, but each emoji seems to be counted as 2 chars by `val().length` while Twitter counts single emoji as a single char.

Please don't be mad if all of this is trivial and stupid, I am more or less a beginner in programming :)

**Comment by xLs51** [comment]
Thank you for the report!

You are right that the `keyup` listener doesn't work on the textarea because we don't write directly into it.
I don't have too much time right now, but I will try to update the library as soon as I can.

You can fix it easily by adding `this.$el.trigger('keyup');` at the end of the `copyTextArea` function, then `$('#twemoji-picker').keyup` should works.

For the second point, you can find plenty of subject about it (for example : http://blog.jonnew.com/posts/poo-dot-length-equals-two).

**Comment by BartoszGancza** [comment]
Thank you very much for your reply!

I tried fixing it on my own by changing the code in `copyTextArea`, but since I am not that well versed in JS I probably did something wrong. This solution is very simple indeed :D

Thank you for the link to that article, it has been a great reference! While searching for some solutions on a character counter for a text box meant to input a tweet text, I have learned that counting characters for tweets is indeed a very complicated issue (mainly due to the way hashtags, @ and links are being handled by Twitter), and requires a hefty amount of additional code, that I cannot fully comprehend yet.
